src/character_attire.cpp

```diff
@@ -1945,6 +1950,101 @@ void outfit::absorb_damage( Character &guy, damage_unit &elem, bodypart_id bp,
+        liquid_descriptor = _( "quite a lot of" );
+    } else if( liquid_remaining <= 125 ) {
+        liquid_descriptor = _( "copious amounts of" );
+    } else if( liquid_remaining <= 150 ) {
+        liquid_descriptor = _( "a cascade of" );
+    } else if( liquid_remaining <= 175 ) {
+        liquid_descriptor = _( "a torrent of" );
+    } else if( liquid_remaining <= 200 ) {
+        liquid_descriptor = _( "a flood of" );
+    } else {
+        liquid_descriptor = _( "a deluge of" );
+    }
+    return liquid_descriptor;
+}
+
+void outfit::splash_attack( Character &guy, const spell &sp, Creature &caster, bodypart_id bp )
+{
+    sub_bodypart_id sbp;
+    sub_bodypart_id secondary_sbp;
+    // if this body part has sub part locations, roll one
+    if( !bp->sub_parts.empty() ) {
+        sbp = bp->random_sub_part( false );
+        // the torso and legs have a second layer of hanging body parts
+        secondary_sbp = bp->random_sub_part( true );
+    }
+    const bool permanent = sp.has_flag( spell_flag::PERMANENT );
+    int intensity = sp.effect_intensity( caster );
+    int liquid_amount = sp.liquid_volume( caster );
+    const int dur_moves = sp.duration( caster );
+    const efftype_id spell_effect = efftype_id( sp.effect_data() );
+    const time_duration dur_td = time_duration::from_moves( dur_moves );
+    // Liquid will splash on the outermost items first.
+    int liquid_remaining = liquid_amount;
+    for( auto iter = worn.rbegin(); iter != worn.rend(); ) {
+        item &armor = *iter;
+        if( !armor.covers( bp ) || armor.has_flag( flag_INTEGRATED ) ) {
+            ++iter;
+            continue;
+        }
+        const std::string pre_damage_name = armor.tname();
+        if( rng( 1, 100 ) <= armor.get_coverage( bp ) && liquid_remaining > 0 ) {
+            guy.add_msg_if_player( m_warning, _( "Your %1s is splashed with %2s of liquid." ),
+                                   armor.tname(), get_liquid_descriptor( liquid_remaining ) );
+            // The item has intercepted the splash to protect its wearer,
+            // now we roll to see if it's affected.
+            // A droplet of bile is less likely to ruin a shirt than a whole bucket.
+            if( rng( 1, 200 - armor.breathability( bp ) ) < liquid_remaining ) {
+                // Apply filth to the item. Currently hardcoded because we don't have other item
+                // flags that would make sense for this. It gets its own probability roll here
+                // because it can't use armor_absorb's.
+                if( sp.has_flag( spell_flag::MAKE_FILTHY ) && !armor.has_flag( flag_INTEGRATED ) &&
+                    !armor.has_flag( flag_SEMITANGIBLE ) && !armor.has_flag( flag_PERSONAL ) &&
+                    !armor.has_flag( flag_AURA ) &&
+                    rng( 1, 200 - armor.breathability( bp ) ) < liquid_remaining ) {
+                    add_msg_if_player_sees( guy, m_bad, _( "%1s %2s is covered in filth!" ), guy.disp_name( true,
```

Comment:
The correct usage of positional arguments is "%1$s` %2$s is covered in filth!".

Reply:
Fixed. Thanks!